csrc/device_lower/pass/scalar_hoist.cpp

```diff
@@ -322,19 +322,25 @@ Val* CommonScalarMap::hoistScalar(
 Val* CommonScalarMap::reuseScalarIfAlreadyComputed(
     Val* value,
     kir::ForLoop* loop) {
+  // Find if value is computed on the host.
+  for (auto val : GpuLower::current()->allKnownVals()) {
+    if (val->sameAs(value)) {
+      return val;
+    }
+  }
```

Comment:
Is this just we forgot to do before? It seems there are some kernel changes. Are they due to this change?

Reply:
The kernel change is due to the change in `Index::iota`, not here. And yes, we just forgot to do this before, but because the "hoist to host" feature is not utilized, so this is just a no-op today.